src/overlays/actors/ovl_Dm_Zl/z_dm_zl.c

```diff
@@ -30,27 +29,268 @@ const ActorInit Dm_Zl_InitVars = {
+    endFrame = (animation->frameCount < 0.0f) ? Animation_GetLastFrame(animation->animation) : animation->frameCount;
+
+    Animation_Change(skelAnime, animation->animation, animation->playSpeed, animation->startFrame, endFrame,
+                     animation->mode, animation->morphFrames);
+}
+
+void DmZl_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    DmZl* this = THIS;
+
+    this->animationIndex = ZELDA_ANIM_FACING_AWAY;
+    this->unk_2BA = 0;
+    this->actor.targetArrowOffset = 1000.0f;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 24.0f);
+    // these three set to NULL should mean they are dynamically allocated
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gZl4Skeleton, NULL, NULL, NULL, 0);
+    DmZl_ChangeAnimation(&this->skelAnime, &sAnimations[this->animationIndex], 0);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = DmZl_DoNothing;
+}
+
+void DmZl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void DmZl_DoNothing(DmZl* this, GlobalContext* globalCtx) {
+}
+
+void DmZl_UpdateCutscene(DmZl* this, GlobalContext* globalCtx) {
+    s32 actionIndex;
+
+    if (Cutscene_CheckActorAction(globalCtx, 0x66)) {
+        actionIndex = Cutscene_GetActorActionIndex(globalCtx, 0x66);
+        if (globalCtx->csCtx.frames == globalCtx->csCtx.actorActions[actionIndex]->startFrame) {
+            s16 nextAnimationIndex = ZELDA_ANIM_FACING_AWAY;
+
+            switch (globalCtx->csCtx.actorActions[actionIndex]->action) {
+                default:
+                case 1:
+                    break;
+                case 2:
+                    nextAnimationIndex = ZELDA_ANIM_TURNING_TOWARD_PLAYER;
+                    break;
+                case 3:
+                    nextAnimationIndex = ZELDA_ANIM_GIVING_OCARINA_START;
+                    break;
+                case 4:
+                    nextAnimationIndex = ZELDA_ANIM_PLAYING_OCARINA_START;
+                    break;
+            }
+
+            if (nextAnimationIndex != this->animationIndex) {
+                this->animationIndex = nextAnimationIndex;
+                DmZl_ChangeAnimation(&this->skelAnime, &sAnimations[this->animationIndex], 0);
+            }
+        }
+
+        Cutscene_ActorTranslateAndYaw(&this->actor, globalCtx, actionIndex);
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+        actionIndex = this->animationIndex;
+
+        if ((actionIndex == ZELDA_ANIM_TURNING_TOWARD_PLAYER) || (actionIndex == ZELDA_ANIM_GIVING_OCARINA_START) ||
+            (actionIndex == ZELDA_ANIM_PLAYING_OCARINA_START)) {
+            // these animations dont loop at the end, they lead into the next animation
+            this->animationIndex++;
+            DmZl_ChangeAnimation(&this->skelAnime, &sAnimations[this->animationIndex], 0);
+        }
+    }
+}
+
+/**
+ * Updates the eye blinking and state, and mouth textures.
+ *  (Copy-Paste from OoT, Unused in MM)
```

Comment:
The function's used, it's called in Update, it's just that most of the control flow isn't touched since `nextEyeState` is never set. You could comment on that above the switch.